tests/queries/0_stateless/01739_index_hint.reference

```diff
@@ -0,0 +1,14 @@
+0	0	0
+0	1	0
```

Comment:
@amosbird but why it returns the first row too for `select * from tbl WHERE indexHint(t = 1) order by t` with `index_granularity=1`?

P.S. it worth converting this test to use `-- { echo }` test hint, see https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/01564_test_hint_woes.sql for example

Reply:
> but why it returns the first row too for select * from tbl WHERE indexHint(t = 1) order by t with index_granularity=1

It's how primary index work in clickhouse. The current granule starts with value 1 doesn't mean all values in previous granule < 1